core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -281,6 +291,12 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
                     });
                     return Errors.NONE;
                 });
+
+                // If we have an acknowledgement completed for a topic-partition, then we should check if
+                // there is a pending share fetch request for the topic-partition and complete it.
+                DelayedShareFetchKey delayedShareFetchKey = new DelayedShareFetchKey(topicIdPartition, groupId);
+                delayedShareFetchPurgatory.checkAndComplete(delayedShareFetchKey);
```

Comment:
It seems that we should call `delayedShareFetchPurgatory` in some other places.
1. When the partition level lock is released.
2. When the HWM advances in a partition.